ports/geckolib/properties.mako.rs

```diff
@@ -628,9 +631,10 @@ fn static_assert() {
     }
 
     fn copy_z_index_from(&mut self, other: &Self) {
-        debug_assert_unit_is_safe_to_copy(self.gecko.mZIndex.mUnit);
+        unsafe { self.gecko.mZIndex.mValue.reset(&mut self.gecko.mZIndex.mUnit) };
         self.gecko.mZIndex.mUnit = other.gecko.mZIndex.mUnit;
         self.gecko.mZIndex.mValue = other.gecko.mZIndex.mValue;
+        unsafe { self.gecko.mZIndex.mValue.addref_opt(&self.gecko.mZIndex.mUnit) };
```

Comment:
z-index doesn't take calc() values, so instead you could assert that the new unit value isn't calc.